packages/react-native/ReactCommon/jsiexecutor/jsireact/JSIExecutor.cpp

```diff
@@ -302,7 +303,7 @@ void JSIExecutor::handleMemoryPressure(int pressureLevel) {
     TRIM_MEMORY_RUNNING_MODERATE = 5,
     TRIM_MEMORY_UI_HIDDEN = 20,
   };
-  const char* levelName;
+  std::string levelName;
```

Comment:
@bwisn That's not how const char * works in C++. Any static string defined here will have a permanent address in the binary, and we can safely pass around pointers to those strings.

The last thing we want to do in this memory callback is allocate more memory on the heap.

Reply:
@javache my bad, you're right  - I thought that the pointer is const, not the variable stored under it. 

But unfortunately, I've tried to set the default value for the levelName ... and it **doesn't** work. It crashes on the << operator again. Below you can see a screenshot from crash and backtrace from lldb. My change with std::string work flawlessly - const char* with default doesn't. 
![obraz](https://github.com/user-attachments/assets/9af0c388-96fe-447c-84ee-3806e091a096)

Reply:
IMO compiler probably does something strange with that const char* while optimizing.

Reply:
What's the pressureLevel in that case? It seems to be trying to run strlen on an empty string, so something's overriding `levelName`. I'm skeptical that the compiler would be incorrectly optimizing this type of basic pattern.

Reply:
@javache I don't have the phone at the moment with me, but as far as I remember, it was the CRITICAL one. I can't see anything in that function what could override the ```levelName```.  And it should be set to default to UNKNOWN (in the switch), so how that would be possible?

Reply:
@javache as far as i can see, the pressureLevel will always be 15 because: 

in file packages/react-native/React/CxxBridge/RCTCxxBridge.mm handleMemoryPressure is called, and the argument is the pressureLevel which is read from static variable
![obraz](https://github.com/user-attachments/assets/5ddb2fd1-f3b2-443f-86d2-7f11e6e0d102)

as you can see here (file packages/react-native/React/Base/RCTConstants.m)

![obraz](https://github.com/user-attachments/assets/504e1de6-788c-4fa2-8f21-5bd3eb50e9fb)

The default value is 15 which means TRIM_MEMORY_RUNNING_CRITICAL and the RCTSetMemoryPressureUnloadLevel is never used anywhere in the code as you can see below.

![obraz](https://github.com/user-attachments/assets/0c1ed457-b6ca-4771-8139-28b72e7b73cf)

Reply:
@javache also, the variable ```static int RCTMemoryPressureUnloadLevel = 15;``` shouldn't be ```static volatile int RCTMemoryPressureUnloadLevel = 15;``` to make sure that compiler won't optimize this if the variable is never changing? At this moment it isn't changing for iOS, but who knows the future ;)